packages/devtools_app/lib/src/screens/performance/performance_controller.dart

```diff
@@ -424,74 +426,101 @@ class PerformanceController extends DisposableController
     _selectedFrameNotifier.value = frame;
 
     if (!offlineController.offlineMode.value) {
-      final bool frameBeforeFirstWellFormedFrame =
-          firstWellFormedFrameMicros != null &&
-              frame.timeFromFrameTiming.start!.inMicroseconds <
-                  firstWellFormedFrameMicros!;
-      if (!frame.isWellFormed && !frameBeforeFirstWellFormedFrame) {
-        // Only try to pull timeline events for frames that are after the first
-        // well formed frame. Timeline events that occurred before this frame will
-        // have already fallen out of the buffer.
-        await processAvailableEvents();
-      }
+      if (useLegacyTraceViewer.value) {
```

Comment:
nit: this if statement is big enough that it might help readability to pull the contents out into 2 helpers that we punt to. i.e.:

```
if(useLegacy){
    _legacyToggleFrame(...);
} else {
    _toggleFrame(...);
}
```